buildkitd.go

```diff
@@ -11,40 +11,60 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/BurntSushi/toml"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 type Buildkitd struct {
 	Addr string
 
+	config  *BuildkitdConfig
 	rootDir string
+	opts    *BuildkitdOpts
 	proc    *os.Process
 }
 
-func SpawnBuildkitd() (*Buildkitd, error) {
+// BuildkitdOpts to provide to Buildkitd
+type BuildkitdOpts struct {
+	ConfigPath string
+}
```

Comment:
I have this purely to allow overriding for testing ... let me know if you have a better idea on how to allow testing the config file generated.